integration/simulation/network/obscuro_node_utils.go

```diff
@@ -116,7 +144,66 @@ func startStandaloneObscuroNodes(params *params.SimParams, stats *stats.Stats, g
 		}
 	}
 
-	return obscuroClients
+	// round-robin the wallets onto the different obscuro nodes, register them each a viewing key
+	walletClients := make(map[string]rpcclientlib.Client)
+	var i int
+	for _, w := range params.Wallets.SimObsWallets {
+		walletClients[w.Address().String()] = rpcclientlib.NewClient(nodeRPCAddresses[i%len(nodeRPCAddresses)])
```

Comment:
@BedrockSquirrel Why are we keying the map with a string, rather than with the wallet itself? More efficient?

Reply:
Tbh I did it because I couldn't remember the Go rules for what could safely be a map key and didn't want any gotchas haha. But also I could imagine being in part of the code that knew the wallet address it wanted to transact from but didn't have easy access to the wallet objects collection.

Can change it if you prefer? (Looks like the wallet struct would be a legit key)